thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/components/AbsoluteChangeRuleDetector.java

```diff
@@ -105,11 +109,44 @@
 
     // make anomalies
     DatasetConfigDTO datasetConfig = data.getDatasetForMetricId().get(me.getId());
-    return DetectionUtils.makeAnomalies(slice, df, COL_ANOMALY, window.getEndMillis(),
+    List<MergedAnomalyResultDTO> anomalies = DetectionUtils.makeAnomalies(slice, df, COL_ANOMALY, window.getEndMillis(),
         DetectionUtils.getMonitoringGranularityPeriod(monitoringGranularity, datasetConfig), datasetConfig);
+    return DetectionResult.from(anomalies, TimeSeries.fromDataFrame(constructAbsoluteChangeBoundaries(df)));
   }
 
   @Override
+  public TimeSeries computePredictedTimeSeries(MetricSlice slice) {
+    InputData data = this.dataFetcher.fetchData(new InputDataSpec().withTimeseriesSlices(this.baseline.scatter(slice)));
+    DataFrame dfBase = this.baseline.gather(slice, data.getTimeseries());
+    return TimeSeries.fromDataFrame(constructAbsoluteChangeBoundaries(dfBase));
+  }
+
+  private DataFrame constructAbsoluteChangeBoundaries(DataFrame dfBase) {
+    if (!Double.isNaN(this.absoluteChange)) {
```

Comment:
Why we have the check here? When will absoluteChange would be NaN?

Reply:
By default, it's NaN which means not set. If it's NaN, no anomaly will be generated.

Reply:
I see. It is a little weird to use NaN as default value.